### ISSUE TITLE
"Systems of PDEs" example in documentation does give the anticipated results

### ISSUE THREAD

**Issue by dohter** [SciML/NeuralPDE.jl#302]
Hello,

I am following the example https://neuralpde.sciml.ai/stable/pinn/system/ and I am not getting the same results in. Mainly there is big discrepancy in u2 and u3 outputs
![sol_u2](https://user-images.githubusercontent.com/23319636/116732582-427cb100-a9a8-11eb-981e-3baaa85f687f.png)
![sol_u3](https://user-images.githubusercontent.com/23319636/116732588-44467480-a9a8-11eb-98d3-f33d0aff329b.png)

And this is the code

```
cd(@__DIR__)
using Pkg
Pkg.activate("../")
using NeuralPDE, Flux, ModelingToolkit, GalacticOptim, Optim, DiffEqFlux

@parameters t, x
@variables u1(..), u2(..), u3(..)
Dt = Differential(t)
Dtt = Differential(t)^2
Dx = Differential(x)
Dxx = Differential(x)^2

eqs = [Dtt(u1(t,x)) ~ Dxx(u1(t,x)) + u3(t,x)*sin(pi*x),
       Dtt(u2(t,x)) ~ Dxx(u2(t,x)) + u3(t,x)*cos(pi*x),
       0. ~ u1(t,x)*sin(pi*x) + u2(t,x)*cos(pi*x) - exp(-t)]

bcs = [u1(0,x) ~ sin(pi*x),
       u2(0,x) ~ cos(pi*x),
       Dt(u1(0,x)) ~ -sin(pi*x),
       Dt(u2(0,x)) ~ -cos(pi*x),
       u1(t,0) ~ 0.,
       u2(t,0) ~ exp(-t),
       u1(t,1) ~ 0.,
       u2(t,1) ~ -exp(-t),
       u1(t,0) ~ u1(t,1),
       u2(t,0) ~ -u2(t,1)]


# Space and time domains
domains = [t ∈ IntervalDomain(0.0,1.0),
           x ∈ IntervalDomain(0.0,1.0)]
# Discretization
dx = 0.1
# Neural network
input_ = length(domains)
n = 8
chain1 = FastChain(FastDense(input_,n,Flux.σ),FastDense(n,n,Flux.σ),FastDense(n,1))
chain2 = FastChain(FastDense(input_,n,Flux.σ),FastDense(n,n,Flux.σ),FastDense(n,1))
chain3 = FastChain(FastDense(input_,n,Flux.σ),FastDense(n,n,Flux.σ),FastDense(n,1))

strategy = GridTraining(dx)
discretization = PhysicsInformedNN([chain1,chain2,chain3], strategy)

pde_system = PDESystem(eqs,bcs,domains,[t,x],[u1,u2,u3])
prob = discretize(pde_system,discretization)

cb = function (p,l)
    println("Current loss is: $l")
    return false
end

res = GalacticOptim.solve(prob,Optim.BFGS(); cb = cb, maxiters=20000)
phi = discretization.phi

using Plots

ts,xs = [domain.domain.lower:dx/10:domain.domain.upper for domain in domains]

initθ = discretization.init_params
acum =  [0;accumulate(+, length.(initθ))]
sep = [acum[i]+1 : acum[i+1] for i in 1:length(acum)-1]
minimizers = [res.minimizer[s] for s in sep]

analytic_sol_func(t,x) = [exp(-t)*sin(pi*x), exp(-t)*cos(pi*x), (1+pi^2)*exp(-t)]
u_real  = [[analytic_sol_func(t,x)[i] for t in ts for x in xs] for i in 1:3]
u_predict  = [[phi[i]([t,x],minimizers[i])[1] for t in ts  for x in xs] for i in 1:3]
diff_u = [abs.(u_real[i] .- u_predict[i] ) for i in 1:3]

for i in 1:3
    p1 = plot(ts, xs, u_real[i], st=:surface,title = "u$i, analytic");
    p2 = plot(ts, xs, u_predict[i], st=:surface,title = "predict");
    p3 = plot(ts, xs, diff_u[i],linetype=:contourf,title = "error");
    plot(p1,p2,p3)
    savefig("sol_u$i")
end
```

**Comment by ChrisRackauckas** [comment]
Do 300 iterations of ADAM before BFGS? And check if you're on the latest version?

**Comment by dohter** [comment]
Thank you for reply!
Here is versioninfo
```
Julia Version 1.6.0
Commit f9720dc2eb (2021-03-24 12:55 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Xeon(R) CPU E3-1505M v6 @ 3.00GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-11.0.1 (ORCJIT, skylake)
```
Also how in this example I can use ADAM before BFGS? Should I build a loss function, train with Flux and then call solve?

**Comment by ChrisRackauckas** [comment]
https://neuralpde.sciml.ai/dev/pinn/fp/ shows one way to do it. 

https://neuralpde.sciml.ai/dev/pinn/2D/ does multiple optimization steps, which if you replace ADAM with BFGS in some, that will do it.

**Comment by dohter** [comment]
Putting ADAM in front brings the loss down to 0.5, but starting from that BFGC does not really pushing it down. Ill play around with learning rates.

**Comment by ChrisRackauckas** [comment]
Are you doing `BFGS(initial_stepnorm=0.01)`?

**Comment by dohter** [comment]
Tried both actually, same result

**Comment by ChrisRackauckas** [comment]
@KirillZubov comments?

**Comment by KirillZubov** [comment]
@dohter I opened the PR. I use QuadratureTraining and add more neurons.  u2 update slower than u1 and u3 very slow updated but in general, prediction converges but slow. 
try to use some of the combinations 
```julia
res = GalacticOptim.solve(prob,ADAM(10^-2); cb = cb, maxiters=10000)
prob = remake(prob,u0=res.minimizer)
res = GalacticOptim.solve(prob,BFGS(); cb = cb, maxiters=200)
...
```
or mayby just ADAM
```julia
res = GalacticOptim.solve(prob,ADAM(10^-2); cb = cb, maxiters=10000)
prob = remake(prob,u0=res.minimizer)
res = GalacticOptim.solve(prob,10^-2; cb = cb, maxiters=10000)
...
```

**Comment by KirillZubov** [comment]
@ChrisRackauckas I suppose that a common problem with the system of pde is that the weights in NN are not updated evenly for each variable, the gradient between the equations is not balanced and the key to the solution should be Adaptive loss reweighting.
We see in the training process that `u3` updates much slower than others since I believe its gradient component is too small relative to the other variable in the equations.

**Comment by ChrisRackauckas** [comment]
Yeah, we need to finish that piece.
